packages/cw1155/README.md

```diff
@@ -0,0 +1,63 @@
+# CW1155 Spec: Multiple Tokens
+
+CW1155 is a specification for managing multiple tokens based on CosmWasm.
+The name and design is based on Ethereum's ERC1155 standard.
+
+Design decisions:
+
+- Fungible tokens and non-fungible tokens are treated equally, non-fungible tokens just have one max supply.
+
+- Approval is set or unset to some operator over entire set of tokens. (More nuanced control is defined in [ERC1761](https://eips.ethereum.org/EIPS/eip-1761), do we want to merge them together?)
+
+- Metadata and token enumeration should be done by indexing events off-chain.
+
+- Mint and burn are mixed with transfer/send messages, otherwise, we'll have much more message types, e.g. `Mint`/`MintToContract`/`BatchMint`/`BatchMintToContract`, etc.
+
+  In transfer/send messges, `from`/`to` are optional, a `None` `from` means minting, a `None` `to` means burning, they must not both be `None` at the same time.
```

Comment:
This seems dangerous to me, as the input is JSON, and extra fields are ignored. So changing `from` to `form` turns a transfer to a mint (likely to fail anyway), but using `too` vs `to` will make you burn your tokens, rather than have the tx rejected by the chain.

@willclarktech and @maurolacy I think we discussed the issue of nullable fields changing functionality to be dangerous. Will especially in the context of CosmJS, some case when we said omitting a field from "ChangeOwner" would make it "ClearOwner" (making no one owner) could be an expensive mistake with a typo.

Can you please comment on this design here (or better look at the [corresponding type definitions](https://github.com/CosmWasm/cosmwasm-plus/pull/247/files#diff-07f199dd55ccb4e6b043c50dded9da214c860437339f9a73e0bea9ed89a85d72R12-R29)) and say if you approve or suggest ways to improve.

Reply:
What about minting/burning if the null address is provided, like Ethereum's `0x0000000000000000000000000000000000000000`? Then at least you have to specify it. Otherwise I'd suggest `mint`/`burn` boolean options to be explicit.